data-prepper-plugins/opensearch/src/test/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/DynamicIndexManagerTests.java

```diff
@@ -0,0 +1,135 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import java.time.format.DateTimeFormatter;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+public class DynamicIndexManagerTests {
+
+    private static final String ID = "id";
+    private static final String DYNAMIC = "dynamic";
+    private static final String INDEX_ALIAS = "test-${" + ID + "}-index-alias";
+    private static final String DATE_PATTERN = "yyyy.MM.dd";
+    private static final String INDEX_ALIAS_WITH_DATE_PATTERN = INDEX_ALIAS+ "-%{" + DATE_PATTERN + "}";
+
+    private IndexManagerFactory indexManagerFactory;
+    private IndexManager indexManager;
+
+    private DynamicIndexManager dynamicIndexManager;
+
+    @Mock
+    private RestHighLevelClient restHighLevelClient;
+
+    @Mock
+    private OpenSearchSinkConfiguration openSearchSinkConfiguration;
+
+    @Mock
+    private ClusterClient cluster;
+
+    @Mock
+    private ClusterGetSettingsResponse clusterGetSettingsResponse;
+
+    @Mock
+    private IndexConfiguration indexConfiguration;
+
+    @Mock
+    private IndicesClient indicesClient;
+
+    @Mock
+    private GetIndexTemplatesResponse getIndexTemplatesResponse;
+
+    @Mock
+    private RestClient restClient;
+
+    @Mock
+    private ResponseException responseException;
+
+    static final String EVENT_TYPE = "event";
+
+    @Before
+    public void setup() throws IOException {
+        initMocks(this);
+
+        indexManagerFactory = new IndexManagerFactory();
+        when(openSearchSinkConfiguration.getIndexConfiguration()).thenReturn(indexConfiguration);
+        when(indexConfiguration.getIsmPolicyFile()).thenReturn(Optional.empty());
+        when(indexConfiguration.getIndexAlias()).thenReturn(INDEX_ALIAS);
+        when(restHighLevelClient.cluster()).thenReturn(cluster);
+        when(cluster.getSettings(any(ClusterGetSettingsRequest.class), any(RequestOptions.class)))
+                .thenReturn(clusterGetSettingsResponse);
+        when(restHighLevelClient.indices()).thenReturn(indicesClient);
+    }
+
+    @Test
+    public void dynamicIndexBasicTest(){
+        when(indexConfiguration.getIndexAlias()).thenReturn(INDEX_ALIAS);
+        when(clusterGetSettingsResponse.getSetting(IndexConstants.ISM_ENABLED_SETTING)).thenReturn("true");
+        String configuredIndexAlias = openSearchSinkConfiguration.getIndexConfiguration().getIndexAlias();
+	try {
+            indexManager = indexManagerFactory.getIndexManager(IndexType.CUSTOM, restHighLevelClient, openSearchSinkConfiguration, configuredIndexAlias);
```

Comment:
This test case is mixing two different behaviors.

* Verify the behavior of `getIndexManager`
* Verifying the behavior of `DynamicIndexManager.getIndexName`

Test cases should only verify a single behavior. Please verify `IndexManagerFactory.getIndexManager` in `IndexManagerFactoryTests`.